### DISCUSSION TITLE
How to set 'read_timeout' for nxos command

### DISCUSSION THREAD

**Discussion by Kani999** [napalm-automation/napalm#1769] [Q&A] [answered]
Hello,

I'm trying to call `get_interfaces()` on `nxos_ssh` driver.
The problem is, that returning interfaces takes too long. 

```
ReadTimeout
Pattern not detected: 'nexus\\\\#' in output.
Things you might try to fix this:
. Explicitly set your pattern using the expect_string argument.
 Increase the read_timeout to a larger value
You can also look at the Netmiko session_log or debug log for more information
```

When I use `netmiko` it works, when I set `read_timeout`
```
from netmiko import ConnectHandler
from getpass import getpass

net_connect = ConnectHandler(
    device_type="cisco_nxos",
    host="",
)
print(net_connect.send_command("show interface", read_timeout=600))
net_connect.disconnect()
```

But I'm not able to bypass `read_timeout` param when using `napalm` command `get_interfaces()`
Is there any way, to pass it to the connection? I tried the `optional_args` and etc. but nothing works. 

Currently, I copied the napalm nxos_ssh method to parse the interfaces, call the command `show interface` via netmiko and give it to the parser. But that's an ugly workaround.

I think this is the place where the param could be set https://github.com/napalm-automation/napalm/blob/5977b148a07c0f75eb3da05c25abe39fa8ece180/napalm/nxos_ssh/nxos_ssh.py#L705

**Comment by ktbyers** [comment, marked as answer]
You should be able to set `read_timeout_override` as an argument specified in NAPALM's optional_args.

Let me know if this fixes your issue.

**Reply by Kani999** [reply]
@ktbyers Thank you, that works.
Is this kind of info in the documentation?

[EDITED]
NVM, found that argument at netmiko docs
